### DISCUSSION TITLE
BlenderBIM Repeated IfcOwnerHistory

### DISCUSSION THREAD

**Discussion by HassanEmam** [IfcOpenShell/IfcOpenShell#1839] [Ideas]
Hello

First I would like to thank all the community members for their tireless contribution to the development, maintenance and improvements to IfcOpenShell.

I have noticed that IfcOwnerHistory is repeated a lot when creating an IFC file in BlenderBIM. The repeated instances are identical and I was thinking from an optimisation view; shouldn't it be better to group these together (the identical ones of course) might achieve better performance and a considerably smaller IFC.

![image](https://user-images.githubusercontent.com/6377734/139331938-57fa4d2c-5f1f-485c-8010-c9fce0c5b9c6.png)

**Comment by Moult** [comment]
Sorry for the late reply but you're absolutely right. The calls to update/create owner history should be batched per transaction, but this has not yet been implemented.
